packages/hdwallet-core/src/wallet.ts

```diff
@@ -137,6 +137,10 @@ export function supportsCosmos(wallet: HDWallet): wallet is CosmosWallet {
   return _.isObject(wallet) && (wallet as any)._supportsCosmos;
 }
 
+export function supportsEthSwitchChain(wallet: HDWallet): wallet is ETHWallet {
```

Comment:
Something feels off about having this as a type guard - it isn't actually a type guard for `ETHWallet`, it's more of a helper with a `boolean` return type.

E.g. if `supportsEthSwitchChain` returns false it does not mean the `HDWallet` object it received is _not_ an `ETHWallet`, only that it does not support ETH switch chain.

That said, I actually don't think any of these functions are _actually_ type guards in any way except for their syntax - but rather mere helper functions.

Probably best to just continue propagating the existing pattern for now...

Reply:
Agreed these should not be type guards to start with. It's not *wrong* per se (if this method returns true, it is an `EthWallet`), it just makes no sense for this to be a type guard, as this discriminates on something entirely different.

You're right consistency > cleanness, we will need to do grep through all the type guards and remove the ones where the guard doesn't actually guard against anything